### ISSUE TITLE
[NBug] Index was out of range. Must be non-negative and less than ...

### ISSUE THREAD

**Issue by MPeti1** [gitextensions/gitextensions#9020]
## Current behaviour

I clicked on an earlier commit of a branch to do and action on it, but before the menu would appear I got this error message.

## Expected behaviour
The program not crashing, but opening the context menu of the selected commit

## Steps to reproduce

This happened with a large repository that only has a few commits, so
1. first create a repository with files of a few hundred MBs
2. commit the changes
3. replace some of the big files, and maybe also rename a few that you haven't actually changed
4. commit again
5. Check out an earlier commit (not branch)
6. Wait a few seconds until the program opens the crash dialog

I can't guarantee that this will happen to you too, because it didn't happen to me always.

GitExtensions is generally slow with big repositories (or with long changelists) (at least when they are on an HDD), and at step 6 you will probably need to wait because of that.
Although if after the crash I reopen the repository in it, then the context menu on said commit appears very quickly, and does not crash.

Please note that I'm aware that in this case I might not be using git for the task it's been made for. My goal is to deduplicate and track changes of some of my directories.
Also, this seemed to be similar to a few other issues, but they were closed as fixed on an earlier version than mine, so I opened a new issue.

## Error Details
```
System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
   at System.Collections.ArrayList.get_Item(Int32 index)
   at System.Windows.Forms.DataGridViewRowCollection.SharedRow(Int32 rowIndex)
   at System.Windows.Forms.DataGridViewRowCollection.get_Item(Int32 index)
   at GitUI.RevisionGridControl.OnGridViewCellMouseDown(Object sender, DataGridViewCellMouseEventArgs e)
   at System.Windows.Forms.DataGridView.OnCellMouseDown(DataGridViewCellMouseEventArgs e)
   at System.Windows.Forms.DataGridView.OnMouseDown(MouseEventArgs e)
   at GitUI.UserControls.RevisionGrid.RevisionDataGridView.OnMouseDown(MouseEventArgs e)
   at System.Windows.Forms.Control.WmMouseDown(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.DataGridView.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
```

## Additional information
git repo with huge files and few commits. this happened after right-clicking a previous commit while waiting for the menu to appear

## Environment
- Git Extensions 3.4.3.9999
- Build d4b0f48bbf3e39a71f5d7ff5231be5678d4aa0f1
- Git 2.28.0.windows.1
- Microsoft Windows NT 10.0.18363.0
- .NET Framework 4.8.4341.0
- DPI 96dpi (no scaling)

**Comment by RussKie** [comment]
Can you provide a public repro?
Also please try the latest dev build to see if it still repros.

**Comment by gerhardol** [comment]
Duplicate of #7374
Corrected in #8534, please try the 3.5 RC1
(There was a correction for #7374 in #7921 that is in 3.4)

**Comment by MPeti1** [comment]
Thank you, I'll try that out.
I would like to help with providing a consistently reproducible case, but currently I'm very low on time, sorry for my late actions.
Also I don't see this time-problem to be solved soon, so if you fell like it you can close the issue, and if you're ok with that I'll reopen it when I can get back to it.
In the meantime, thank you for the patch, I'll try to test it in a few days.

**Comment by RussKie** [comment]
Presume closed in 3.5.
If not, please holler back to reopen.
